### PR TITLE
chore: move redirect test server to rust

### PR DESCRIPTION
**Pull request by ry**

### CODE REVIEW

File: test_util/src/lib.rs [hunk 1]
```diff
@@ -58,42 +69,133 @@ pub fn deno_exe_path() -> PathBuf {
...
+          "Content-Type",
+          "application/javascript",
+        )
+      } else if p.ends_with(".json") {
+        warp::reply::with_header(reply, "Content-Type", "application/json")
+      } else {
+        // TODO how to I just reply with identity?
+        warp::reply::with_header(reply, "X-Foo", "TODO")
+      }
+    });
```
Comment (bartlomieju) [inline test_util/src/lib.rs:145]:
```suggestion
.map(|path: warp::path::Peek, f: warp::fs::File| {
      let p = path.as_str();
      if p.ends_with(".ts") || p.ends_with(".tsx") {
        warp::reply::with_header(
          f,
          "Content-Type",
          "application/typescript",
        ).into_response()
      } else if p.ends_with(".js") || p.ends_with(".jsx") {
        warp::reply::with_header(
          f,
          "Content-Type",
          "application/javascript",
        ).into_response()
      } else if p.ends_with(".json") {
        warp::reply::with_header(f, "Content-Type", "application/json").into_response()
      } else {
        f.into_response()
      }
```

**Comment by piscisaureus** [comment]
@ry I'm giving up on fixing the benchmarks - I have no idea what is happening here and why (it seems that) every benchmark starts a dozen servers. I'm no longer surprised that CI is red anyway, this is a mess.

Feel free of course to roll back all the non-fixes I've pushed.

File: Cargo.lock [hunk 2]
```diff
@@ -2455,14 +2481,26 @@ dependencies = [
...
+
 [[package]]
 name = "tokio-rustls"
 version = "0.13.1"
 source = "registry+https://github.com/rust-lang/crates.io-index"
 checksum = "15cb62a0d2770787abc96e99c1cd98fcf17f94959f3af63ca85bdfb203f051b4"
 dependencies = [
  "futures-core",
- "rustls",
+ "rustls 0.17.0",
```
Comment (bartlomieju) [inline Cargo.lock:2503]: I think it means we got two different versions of `tokio-rustls` and `rustls`

File: test_util/src/lib.rs [hunk 3]
```diff
@@ -54,27 +70,362 @@ pub fn deno_exe_path() -> PathBuf {
...
   fn drop(&mut self) {
-    match self.child.try_wait() {
+    match self.test_server.try_wait() {
       Ok(None) => {
-        self.child.kill().expect("failed to kill http_server.py");
-      }
-      Ok(Some(status)) => {
-        panic!("http_server.py exited unexpectedly {}", status)
+        self.test_server.kill().expect("failed to kill test_server");
+        let _ = self.test_server.wait();
```
Comment (bartlomieju) [inline test_util/src/lib.rs:420]: 👍  good to know

**Comment by piscisaureus** [comment]
@ry not very important I suppose since this is not performance critical code.
Should you ever find yourself not wanting to allocate so many boxes, this is the easier way -> https://github.com/denoland/deno/pull/6364/commits/bbb0ccb83847f6eb09ddc3c322031486f1c09a60

File: Cargo.lock (hunk 2 above)
Comment (piscisaureus) [reply Cargo.lock:2503]: Yeah I found that weird too but there does not seem to be an easy workaround.
IIRC (from last night) reqwest requires the latest version of tokio-rustls, and warp is a little behind in updating it's dependencies...

File: cli/tests/echo_server.ts [hunk 4]
```diff
@@ -1,4 +1,4 @@
-const addr = Deno.args[0] || "0.0.0.0:4544";
+const addr = Deno.args[1] || "0.0.0.0:4544";
```
Comment (piscisaureus) [inline cli/tests/echo_server.ts]: Of course you don't have to include this but I also don't understand why not. 🤷‍♀️ 
I mean this is just a clear typo...

**Pull request merged**